### DISCUSSION TITLE
General Discussion Thread

### DISCUSSION THREAD

**Discussion by MadByteDE** [togfoxy/MarsLander#187] [General]
Here is a discussion for general topics. :gift:  
(I believe atm we don't need to open new discussions for each small topic we have)

I think we should set a max amount of PRs at once like 2. The issues we had today made me see that having too much going is really bad since at all fronts merge conflicts arise and we are jumping from one PR to the next which is a lot of work for everyone.

It doesn't have to be a rule, but maybe something everyone has in mind working on stuff. It shouldn't be a problem to wait with the own PR until others have been resolved. There always will be merge conflicts, but trying to solve them in a single PR is much easier then doing this for two or three at the same time.

**Comment by Philbywhizz** [comment]
I have cancelled my PR as it contained some errors in it (I commited a merge conflict into it which was causing confusion). I'll address it at a later date.

I agree with your comments however and the great work you have done with reorganizing and separating some of the code (like lander, modules, etc) will make it easier.

Bite sized changes are the best.

**Comment by MadByteDE** [comment]
**Just a quick heads-up:**
The implementation of the assetloader led me to redo more stuff then I thought since I had to touch so many parts of the code. 
Tt looks like I'll be able to sent in the PR today, so it might be helpful if you guys could hold onto your PRs until my changes have been resolved. It's likely that there will be conflicts. Sorry!

**Reply by togfoxy** [reply]
Go nuts!  :)

I have rewritten the socket stuff but it is isolated in a new file so there will be zero conflict.

**Reply by Philbywhizz** [reply]
Don't apologise! All good.

**Comment by togfoxy** [comment]
I've made two PR's but I've documented in the PR the expected sequence:

PR order:

    asset loader
    enet
    rangefinder fix



There will be conflicts but I know what they are. I keep fixing the same bugs in every branch so resolving will only take a minute.

**Reply by MadByteDE** [reply]
Awesome, when those PRs are resolved I can finally start to implement the new lander physics stuff :star_struck:  Can't wait!
Maybe we should write down some ideas for what is coming next for v0.11 and even v0.12? Would be cool to have some more gameplay features for the next release because the last one was almost a restructure / bugfixing release. Some things I've seen so far:
- Multiplayer improvements (chat, score display, notifications)
- Parachutes ?

For the "Player 'togfoxy' joined" type of notifications we could work on a proper notification system as well.

Also we might need to work through our backlog of issues, they are getting more and more every day :smile:

**Comment by togfoxy** [comment]
I suspect we'll get more momentum once v0.11 drops.

**Comment by MadByteDE** [comment]
I want to get rid of that thought so I just drop it here quickly and then go to bed.

I feel like I'm to impulsive with some decisions I make like what to change in the code or on Github. That likely resulted in #194 being as big as it is and touching more stuff then we agreed on.

I'm thankful that you guys are so open and considerate, but I really thing I have to change that and turn it down a notch for the good teamwork to function in the long run.

Whenever you think I'm doing to much, in a PR for example, just be open and let me know. I'm not easily offended or anything, so don't hold back :thumbsup:

My next goal is to get the lander physics in and after that I'd look into the existing issues and try to work through some older ones first. I'll let you know when I've something I think is important and we can discuss if we want me to work on it.

**Reply by togfoxy** [reply]
Community projects means working with other people and that means hard work. :)

Your PR was a big one for obvious reasons and it was something that simple had to happen. Thanks for all that time. I hope you got something from it. :)

Our 'lesson learned' is smaller PR's are preferred because the big PR's become blockers. That couldn't be avoided in this case. We may need more blocking PR's in the future. The thing that worked well this time is us calling out the big change and holding off on the light merges.

I'm not sure your preferred style but did you want to create a millstone for the new physics and put all your ideas in there? You can then work through ideas one at a time and stay on topic. That's my method but up to you. If that's not your style then don't do it. :)

**Reply by MadByteDE** [reply]
> I'm not sure your preferred style but did you want to create a millstone for the new physics and put all your ideas in there? You can then work through ideas one at a time and stay on topic. That's my method but up to you.

I'll try something like this yeah. thanks.

**Comment by togfoxy** [comment]
@MadByteDE 

I had some free time at lunch so I set up a new project for your physics thing because you seem pretty keen to get cracking on that :)

https://github.com/togfoxy/MarsLander/projects/3

I create a project with some issues to help break down your ideas into small chunks.

**IF YOU DON'T LIKE THIS PROJECT OR THESE ISSUES THEN WE CAN DELETE IT.**

This is a suggestion and only a suggestion. This is my workflow. This is what works for me. If this is not helpful for you then we'll just delete it.  :)  It's that simple.

If you like it, or would like to try it, then I suggest you use the project and issues board to organise your thoughts, code and do PR's as each issue is addressed. You don't have to wait for the whole project to be finished - but of course - you can if you want to.

These are just suggestions. I'm not sure what workflow works best for you.

**Comment by MadByteDE** [comment]
My record (and a tiny bug) without cheating :smile: 
![2021-11-09_21-02](https://user-images.githubusercontent.com/4752038/140996367-68adf2e0-d5f2-4904-ab91-8a7c7118e567.png)

**Reply by togfoxy** [reply]
V nice! I think I get about 50,000 so that's a great score.

**Reply by MadByteDE** [reply]
The trick is to play normally as long as possible and then rocket to the stars until fuel is out :laughing:

**Comment by MadByteDE** [comment]
Head-up:
I think we need to push the v.0.11 release a bit out because of [this bug](https://github.com/togfoxy/MarsLander/pull/234/files#r746720287).
Since I've never reverted a merge, I'd try to fix it up now. If I can't find a clean solution I'll wait for you guys to think about something tomorrow.

**Comment by MadByteDE** [comment]
Okay guys, since I was pretty much able to work without risking to break pending PRs, I pushed my commits directly to the repo to fix game breaking stuff as fast as possible in preparation for release v.0.11. 

Additional changes (not bug fixes):
`made smoke.lua less dependent on lander.lua` 862770a9b8c87ae326c8f13a7d4e1309cbed3fdf
`Assetloader: Add option to toggle debug logging` 75ba178fafa8e29cb244b5ca8a7a8978da9d189c
`Remove duplicate of the same process in main.lua` b5b121300b69eddf2adc50720567581ed293e8db

Please play-test and push bugfixes if needed. 

Also, not forget to pull the changes into your local repos. If I caused any conflicts with WIP code, feel free to link me in the upcoming PR and I'll happily fix the merge conflicts for causing the trouble :grin: :heart:

**Comment by togfoxy** [comment]
With v0.11 out now (thanks for your help!) I'm going to take a few days break and intentionally not visit github.  I'll be back next in a few days time to help with whatever needs helping. :)

**Comment by togfoxy** [comment]
I've created a new project call 'Backlog'. It's an Agile term that means a place to put all the good ideas that are not a priority right now. I've put all the things that we know aren't going to happen soon - but - if you want to work on something from the backlog then you can move it from the backlog to the "normal" project.

I've simply split out the issues because the main project was getting cluttered.

**Comment by togfoxy** [comment]
Anyone know why my parachute branch is showing differences when compared with MAIN branch even though the code is 90% the same and in sync?

e.g. Parachute branch and main branch shows the same code:
`local TIMER_HOST_SEND_INTERVAL = 0.04`

but when I do a PR compare it shows a difference from 0.05 -> 0.04. I use to think my workflow was messed up but the code base is the same so idk what's going on with that compare. You should be able to see it yourself in github webpage and check if I'm crazy.

**Comment by togfoxy** [comment]
IDK what is going on. I created a whole new branch and uploaded the same code into it. The PR compare was perfect. Exactly as expected. I'm just going to create a new branch next time I think something is fishy.

**Reply by MadByteDE** [reply]
What does your git workflow look like?

I use [GitKraken](https://www.gitkraken.com/) to separate repos for all my branches to have independent code bases. It's pretty awesome because it always shows you if something changed and let you revert everything very easily by GUI. There are a lot of extra features in there for QoL as well.

In your case I'd say either you forgot to push the latest changes to your branch or it's some whitespace issue again maybe.
